### DISCUSSION TITLE
Is there any solution to remember third-party switches even when safe mode is on?

### DISCUSSION THREAD

**Discussion by Magic-wei** [BryanJenksCommunity/FAQ#33] [Q&A] [answered]
Hi Bryan,

Thank you for your amazing explanation about Obsidian! I have a question about safe mode. 

I'm using Obsidian on Ubuntu 16.04 LTS/Windows 10 and storing my vault on Github. The problem is, every time I clone and open this vault in Obsidian, it will automatically switch to safe mode which also turns off all third-party plugins (and some configurations are set to default too), so I have to turn on all these plugins manually. It's okay since this won't happen that often but it doesn't feel good because I'm using many third-party plugins and I just can't remember all the details of their configurations. Is there any solution to store the safe mode status and keep it off when we clone the vault from Github and open it? Or is it possible to remember third-party switches even when safe mode is on?

Best regards,
Wei

**Comment by tallguyjenks** [comment, marked as answer]
So here are some ideas,

- Safemode will always default to off for security reasons, but the settings/configs/keybindings don't need to be remade every time
- if you use github then i personally add every file in the vault to version control excent the `.obsidian/trash` and `.obsidian/cache` directories which only contain junk i don't need
  + in some of those files are the configuration settings so that once you turn safe mode off your settings are still the same as they are in your home vault
- Another idea is to not repeatedly clone the repo, and to instead just pull the changes but if that's not possible the aforementioned ideas should work

are you using my github workflow for automating the management of your vault/repo?

**Reply by tallguyjenks** [reply]
tl;dr, yes, there's some files in `.obsidian` to manage the settings that can be included in version control and when pulled down and safemode turned off, will retain your settings with no headache

**Reply by Magic-wei** [reply]
Thank you for your reply! Apologize first because I mistook safe mode on and off. By default safe mode is on but if we want to use third-party plugins we need it to be off. 

I haven't started using your automatic management workflow yet but I did read your blog and followed some of your Obsidian and gitignore configurations. And yes you're right we shouldn't repeatedly clone the repo, actually we only need to clone to a new computer if we changed our computer. So basically it's just fine so far.

The problem is, safe mode will be automatically turned on every time we clone and open the vault (will get the image below, need to click `Turn off Save Mode` and recover third-party plugins manually later). At the meantime, config files will be changed as well. So looks like the config files maintained on Github can't help in this case. Perhaps we can copy those config files to a different path and copy back after safe mode turned off.
![safe_mode_issue](https://user-images.githubusercontent.com/29951097/108605561-be43f700-7382-11eb-8f57-9ffbb6993975.png)

**Reply by Magic-wei** [reply]
Just tried:
1. Back up `.obsidian/config` and `.obsidian/workspace` before we clone and open the vault for the first time. 
2. Turn off Safe Mode. 
3. Overwrite these two files with backup. Everything will be exactly what it was.

**Reply by tallguyjenks** [reply]
im unsure why your configs are reset when you pull the repo down. When i pull my repo down fresh, after i turn safe mode off, re-activate all the community plugins, all of my settings are exactly how they were on my home machine/vault.

for the repo i literally add EVERYTHING except `.obsidian/trash` and `.obsidian/cache` and i have no issues, so I'm unsure what the error might be :( maybe if you try out my workflow that may resolve the issue but at this point I'm unsure

**Reply by Magic-wei** [reply]
Not all configs are reset, I noticed two options changed and maybe I just remembered wrong :joy:. Re-activating the community plugins works but I just don't want to do this step either so I tried my solution which works for me. Thanks again for your discussion! I think this issue can be closed now. Have a nice weekend!
